src/index.ts

```diff
@@ -0,0 +1,62 @@
+// FIXME: lint fails
+/* eslint-disable no-underscore-dangle */
+import http = require('http');
+import https = require('https');
+
+import { got as gotCjs } from 'got-cjs';
+
+// @ts-expect-error Missing types
+import HeaderGenerator = require('header-generator');
+
+import { TransformHeadersAgent } from './agent/transform-headers-agent';
+import { SCRAPING_DEFAULT_OPTIONS } from './scraping-defaults';
+
+import { optionsValidationHandler } from './hooks/options-validation';
+import { customOptionsHook } from './hooks/custom-options';
+import { browserHeadersHook } from './hooks/browser-headers';
+import { proxyHook } from './hooks/proxy';
+import { http2Hook } from './hooks/http2';
+import { insecureParserHook } from './hooks/insecure-parser';
+
+const got = gotCjs.extend({
+    mutableDefaults: true,
+    ...SCRAPING_DEFAULT_OPTIONS,
+    context: {
+        headerGenerator: new HeaderGenerator(),
+        useHeaderGenerator: true,
+        insecureHTTPParser: true,
+    },
+    agent: {
+        http: new TransformHeadersAgent(http.globalAgent) as unknown as http.Agent,
+        https: new TransformHeadersAgent(https.globalAgent) as unknown as https.Agent,
+    },
+    hooks: {
+        init: [
+            optionsValidationHandler,
+            customOptionsHook,
+        ],
+        beforeRequest: [
+            http2Hook,
+            proxyHook,
+            browserHeadersHook,
+            insecureParserHook,
+        ],
+    },
+});
+
+export * from 'got-cjs';
+export default got;
+
+export { got };
```

Comment:
Holy smokes... there's 5 different exports here. Personally, I think it'd be cleaner for a major release to export it as an object `export { got }` and thats all. Would make all of this much cleaner, and compatible with CJS

Reply:
I agree. /cc @mnmkng

Reply:
I have only a faint idea how this ESM vs CJS vs TS build arguments exporting magic works. So yeah, I'm definitely for a cleaner API if that can be done. We'll have to address it in the SDK, if the imports will be different, but that's fine.

Reply:
I think it'd be best for everyone if we just `export { got }`. That's a major bump for this module, and a patch bump for modules using this module, as well as keeping code consistent for CJS, ESM and TS 👍